core/src/serve_repair.rs

```diff
@@ -323,31 +327,35 @@ impl ServeRepair {
         requests_receiver: &PacketBatchReceiver,
         response_sender: &PacketBatchSender,
         stats: &mut ServeRepairStats,
-        packet_threshold: &mut DynamicPacketToProcessThreshold,
+        data_budget: &DataBudget,
     ) -> Result<()> {
         //TODO cache connections
         let timeout = Duration::new(1, 0);
         let mut reqs_v = vec![requests_receiver.recv_timeout(timeout)?];
+        const MAX_REQUESTS_PER_ITERATION: usize = 1024;
         let mut total_requests = reqs_v[0].len();
 
-        let mut dropped_requests = 0;
         while let Ok(more) = requests_receiver.try_recv() {
             total_requests += more.len();
-            if packet_threshold.should_drop(total_requests) {
-                dropped_requests += more.len();
-            } else {
-                reqs_v.push(more);
+            reqs_v.push(more);
+            if total_requests > MAX_REQUESTS_PER_ITERATION {
+                break;
```

Comment:
Wouldn't this cause the channel to grow indefinitely with unprocessed packets?
Old code was draining the channels and discarding unprocessed packets which would prevent this.

Reply:
Yeah that's true, I can reintroduce the drain + drop logic